aws-storage-s3/src/test/java/com/amplifyframework/storage/s3/StorageComponentTest.java

```diff
@@ -0,0 +1,235 @@
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.storage.s3;
+
+import android.content.Context;
+
+import com.amplifyframework.AmplifyException;
+import com.amplifyframework.storage.StorageCategory;
+import com.amplifyframework.storage.StorageCategoryConfiguration;
+import com.amplifyframework.storage.StorageException;
+import com.amplifyframework.storage.StorageItem;
+import com.amplifyframework.storage.result.StorageDownloadFileResult;
+import com.amplifyframework.storage.result.StorageListResult;
+import com.amplifyframework.storage.result.StorageRemoveResult;
+import com.amplifyframework.storage.result.StorageUploadFileResult;
+import com.amplifyframework.storage.s3.service.StorageService;
+import com.amplifyframework.testutils.Await;
+import com.amplifyframework.testutils.RandomString;
+
+import com.amazonaws.mobileconnectors.s3.transferutility.TransferListener;
+import com.amazonaws.mobileconnectors.s3.transferutility.TransferObserver;
+import com.amazonaws.mobileconnectors.s3.transferutility.TransferState;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+
+import java.io.File;
+import java.util.Collections;
+import java.util.Date;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.anyString;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Test that storage category correctly invokes the methods via
+ * AWSS3StoragePlugin.
+ */
+@RunWith(RobolectricTestRunner.class)
+public final class StorageComponentTest {
+
+    private StorageCategory storage;
+    private StorageService storageService;
+
+    /**
+     * Sets up Storage category by registering a mock AWSS3StoragePlugin
+     * instance to Amplify and configuring.
+     * @throws AmplifyException if Amplify fails to configure with mock
+     *         Storage category configuration.
+     */
+    @Before
+    public void setup() throws AmplifyException {
+        this.storage = new StorageCategory();
+        this.storageService = mock(StorageService.class);
+        StorageService.Factory storageServiceFactory = (context, region, bucket) -> storageService;
+        IdentityIdProvider identityIdProvider = RandomString::string;
+        this.storage.addPlugin(new AWSS3StoragePlugin(storageServiceFactory, identityIdProvider));
+        this.storage.configure(buildConfiguration(), mock(Context.class));
+    }
+
+    private static StorageCategoryConfiguration buildConfiguration() {
+        StorageCategoryConfiguration configuration = new StorageCategoryConfiguration();
+        JSONObject storageJson;
+        try {
+            storageJson = new JSONObject()
```

Comment:
I realized later, this might as well be:
```
try {
    configuration.populateFromJson(new JSONObject()
````
without all the extra ceremony (currently) to keep the `JSONObject` at method scope.